In [50]:
## creamos el objeto dataf y cargamos los datos en él
setwd("../data")
df <- read.csv("banco_mundial_tierras_cultivables_has_por_persona.csv", 
                  sep =";", skip = 3,  stringsAsFactors= FALSE, dec= ",")

In [44]:
class(df)

[1] "data.frame"

In [45]:
dim(df) # comprobamos el tipo y el tamaño del objeto creado

[1] 264  59

El objeto creado tiene **264 observaciones** (filas) y **59 variables(columnas)**

In [46]:
names(df) # devuelve el nombre de las variables

[1] "Country.Name" "Country.Code" "X1960"        "X1961"        "X1962"       
 [6] "X1963"        "X1964"        "X1965"        "X1966"        "X1967"       
[11] "X1968"        "X1969"        "X1970"        "X1971"        "X1972"       
[16] "X1973"        "X1974"        "X1975"        "X1976"        "X1977"       
[21] "X1978"        "X1979"        "X1980"        "X1981"        "X1982"       
[26] "X1983"        "X1984"        "X1985"        "X1986"        "X1987"       
[31] "X1988"        "X1989"        "X1990"        "X1991"        "X1992"       
[36] "X1993"        "X1994"        "X1995"        "X1996"        "X1997"       
[41] "X1998"        "X1999"        "X2000"        "X2001"        "X2002"       
[46] "X2003"        "X2004"        "X2005"        "X2006"        "X2007"       
[51] "X2008"        "X2009"        "X2010"        "X2011"        "X2012"       
[56] "X2013"        "X2014"        "X2015"        "X2016"

**read.csv()** hace uso de **make.names()** para comprobar la validez de los nombres de las etiquetas. Estas reglas exigen que un nombre válido debe de comenzar por una letra. Debido a eso R añade una "X" a las etiquetas de las variables. Esto puede ser desactivado usando **check.names = FALSE**

In [36]:
object.size(df) # devuelve el tamaño del objeto en memoria

156456 bytes

In [47]:
head(df)

Country.Name   Country.Code X1960 X1961      X1962      X1963      X1964     
1 Aruba          ABW          NA    0.03607634 0.03557137 0.03527648 0.03506803
2 Afganistán     AFG          NA    0.83453659 0.82389351 0.81288414 0.80153228
3 Angola         AGO          NA    0.46410375 0.46027479 0.45481778 0.45131382
4 Albania        ALB          NA    0.26027232 0.25477424 0.24962825 0.24419351
5 Andorra        AND          NA    0.06956522 0.06506181 0.06093103 0.05724426
6 El mundo árabe ARB          NA    0.48375236 0.47091441 0.46038746 0.45144367
  X1965      X1966      X1967      ... X2007      X2008      X2009     
1 0.03486750 0.03465304 0.03445009 ... 0.01975894 0.01973301 0.01971356
2 0.78483348 0.77174395 0.75622094 ... 0.29282267 0.28555694 0.27827839
3 0.44653659 0.44375627 0.44115389 ... 0.16192260 0.15625417 0.17738715
4 0.24667644 0.24809709 0.24827050 ... 0.19461168 0.20696811 0.20802598
5 0.05391126 0.05089836 0.04817420 ... 0.02781708 0.02623389 0.02723118
6 0.44333417 0.42989788 0.42031216 ... 0.17563888 0.16892018 0.16519897
  X2010      X2011      X2012      X2013      X2014      X2015 X2016
1 0.01967168 0.01959766 0.01949755 0.01938229 0.01926875 NA    NA   
2 0.27056053 0.26224730 0.25377107 0.24533835 0.23722435 NA    NA   
3 0.17544512 0.19406600 0.18727972 0.18847357 0.18201765 NA    NA   
4 0.21489718 0.21409923 0.21345324 0.21315385 0.21307644 NA    NA   
5 0.02960367 0.03104441 0.02911526 0.03589642 0.03534327 NA    NA   
6 0.16195985 0.15821628 0.15573721 0.15425933 0.15205697 NA    NA

In [54]:
summary(df) # resumen de la estructura del objeto

 Country.Name       Country.Code        X1960             X1961        
 Length:264         Length:264         Mode:logical   Min.   :0.00176  
 Class :character   Class :character   NA's:264       1st Qu.:0.15235  
 Mode  :character   Mode  :character                  Median :0.32391  
                                                      Mean   :0.39022  
                                                      3rd Qu.:0.51495  
                                                      Max.   :3.29834  
                                                      NA's   :48       
     X1962             X1963             X1964             X1965        
 Min.   :0.00229   Min.   :0.00223   Min.   :0.00217   Min.   :0.00211  
 1st Qu.:0.15113   1st Qu.:0.15041   1st Qu.:0.14576   1st Qu.:0.14410  
 Median :0.32006   Median :0.31624   Median :0.31102   Median :0.30643  
 Mean   :0.38491   Mean   :0.38061   Mean   :0.37576   Mean   :0.37055  
 3rd Qu.:0.50552   3rd Qu.:0.50641   3rd Qu.:0.49717   3rd 

y como, muchas columnas contienen datos no validos (NA) e incluso algunas no tienen ningún dato (X1960, X2015, X2016)

In [55]:
str(df) # estructura del objeto

'data.frame':	264 obs. of  59 variables:
 $ Country.Name: chr  "Aruba" "Afganistán" "Angola" "Albania" ...
 $ Country.Code: chr  "ABW" "AFG" "AGO" "ALB" ...
 $ X1960       : logi  NA NA NA NA NA NA ...
 $ X1961       : num  0.0361 0.8345 0.4641 0.2603 0.0696 ...
 $ X1962       : num  0.0356 0.8239 0.4603 0.2548 0.0651 ...
 $ X1963       : num  0.0353 0.8129 0.4548 0.2496 0.0609 ...
 $ X1964       : num  0.0351 0.8015 0.4513 0.2442 0.0572 ...
 $ X1965       : num  0.0349 0.7848 0.4465 0.2467 0.0539 ...
 $ X1966       : num  0.0347 0.7717 0.4438 0.2481 0.0509 ...
 $ X1967       : num  0.0345 0.7562 0.4412 0.2483 0.0482 ...
 $ X1968       : num  0.0343 0.7397 0.4384 0.2472 0.0457 ...
 $ X1969       : num  0.0341 0.7241 0.4366 0.245 0.0434 ...
 $ X1970       : num  0.0339 0.7073 0.428 0.244 0.0412 ...
 $ X1971       : num  0.0336 0.6919 0.4186 0.2422 0.0391 ...
 $ X1972       : num  0.0334 0.6748 0.4087 0.2407 0.0372 ...
 $ X1973       : num  0.0332 0.6576 0.3985 0.2386 0.0354 ...
 $ X1974

In [60]:
colMeans(df[3:5,], na.rm = TRUE)

ERROR: Error in colMeans(df[3:5, ], na.rm = TRUE): 'x' debe ser numérico
